Source/Reporters/DataTableReporter.jsl

```diff
@@ -0,0 +1,140 @@
+﻿// SPDX-License-Identifier: Apache-2.0
+
+Names Default To Here( 0 );
+
+/*
+	Class: UtDataTableReporter
+		---Prototype---
+		class UtDataTableReporter inherits UtReporter
+		---------------
+		Saves results from tests as rows in a JMP table
+*/
+Define Class(
+	"UtDataTableReporter",
+	Base Class( "UtReporter" ),
+	
+	//Table to store results
+	dt = .;
+	
+	_init_ = Method({}, 
+		this:dt = New Table( "Test Results",
+			visibility("private"),
+			Add Rows( 0 ),
+			New Script(
+				"Summary",
+				Tabulate(
+					Show Control Panel( 0 ),
+					Add Table( Row Table( Grouping Columns( :result ) ) )
+				)
+			),
+			New Column( "DateTime",        Numeric,    "Continuous", Format( "m/d/y h:m:s", 23, 0 ),
+				Input Format( "m/d/y h:m:s", 0 ), Set Values([]) ),
+			New Column( "label",           Expression, "None",    Set Values({}) ),
+			New Column( "Case Name",       Character,  "Nominal", Formula( Eval List( :label )[1] ) ),
+			New Column( "Test Name",       Character,  "Nominal", Formula( Eval List( :label )[2] ) ),
+			New Column( "Assertion Label", Character,  "Nominal", Formula( Eval List( :label )[3] ) ),
+			New Column( "Assertion ID",    Numeric,    "Nominal", Formula( Eval List( :label )[4] ) ),
+			New Column( "test expr",       Expression, "None",    Set Values({}) ),
+			New Column( "description",     Character,  "Nominal", Set Values({}) ),
+			New Column( "lre",             Expression, "None",    Set Values({}) ),
+			New Column( "payload",         Expression, "None",    Set Values({}) ),
+			New Column( "result",          Character,  "Nominal", Set Values({}) )
+		);
+		1;
+	);
+
+	// Function: Table
+	// Returns a reference to the results data table
+	Get Table Reference = function = Method( {}, this:dt );
+	
+	// Function: Table Data View
+	// Returns a reference to a new view of the data table
+	New Data View = function = Method( {},
+		this:dt << New Data View;
+	);
+	
+	// Function: add expression failure
+	add expression failure = Method( {label, payload=Empty()},
+		rw = { 
+			DateTime = today(),
+			label= label,
+			test expr = "is and expression",
+			description = "was not an expression",
+			lre = Empty(),
+			payload = Name Expr( payload ),
+			result = "expression failure"
+		};
+		this:dt << Add Rows( rw );
+	);
+
+	// Function: add failure
+	add failure = Method( { label, test expr, description, mismatch, lre, payload=Empty() },	
```

Comment:
I really think you should capture the `mismatch`. It is really valuable for diagnosing failures.